tests/test_dataset.py

```diff
@@ -612,7 +608,19 @@ def test_group_only_for_single_group(self):
         ds = Dataset(subset_index=_create_valid_index())
 
         with pytest.raises(ValueError, match="only a single group left"):
-            _ = ds.group
+            _ = ds.group_label
+
+    def test_datapoint_label(self):
```

Comment:
We should add a test for grouped labels.

Ideally just parameterize this test with different groupby modes

Reply:
I added some tests for `datapoint_labels` and `group_labels`, can you check if that covers what you meant here?